### PR TITLE
build: upgrade device-sdk-go

### PR DESCRIPTION
**Pull request by cloudxxx8**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: Fix https://github.com/edgexfoundry/device-virtual-go/issues/130

## What is the new behavior?
upgrade device-sdk-go to v1.2.4-dev.13

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module github.com/edgexfoundry/device-virtual-go
 
 require (
-	github.com/edgexfoundry/device-sdk-go v1.2.3
+	github.com/edgexfoundry/device-sdk-go v1.2.4-dev.12
```
Comment (lenny-intel) [inline go.mod:4]: The upgrate to latest SDk should be in a separate PR from enabling DependaBot.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (cloudxxx8) [reply go.mod:4]: removed dependabot from this PR

**Comment by FelixTing** [comment]
I did a quick test and ran into the following two panic errors when service started.

When launching the device-virtual service for the first time:
> level=INFO ts=2020-09-18T02:21:20.486262Z app=device-virtual source=config.go:193 msg="Loaded configuration from ./res/configuration.toml"
> level=ERROR ts=2020-09-18T02:21:20.487497Z app=device-virtual source=logger.go:74 msg="logTarget cannot be blank, using stdout only"
> level=INFO ts=2020-09-18T02:21:20.487518Z app=device-virtual source=config.go:304 msg="Using local configuration from file (0 envVars overrides applied)"
> level=INFO ts=2020-09-18T02:21:20.487563Z app=device-virtual source=httpserver.go:99 msg="Web server starting (localhost:49990)"
> level=INFO ts=2020-09-18T02:21:20.487642Z app=device-virtual source=init.go:144 msg="Check Metadata service's status by ping..."
> level=INFO ts=2020-09-18T02:21:20.48768Z app=device-virtual source=init.go:144 msg="Check Data service's status by ping..."
> level=INFO ts=2020-09-18T02:21:20.492322Z app=device-virtual source=init.go:68 msg="Service clients initialize successful."
> level=INFO ts=2020-09-18T02:21:20.494172Z app=device-virtual source=service.go:146 msg="Device Service device-virtual doesn't exist, creating a new one"
> level=INFO ts=2020-09-18T02:21:20.494514Z app=device-virtual source=service.go:205 msg="Addressable device-virtual doesn't exist, creating a new one"
> level=INFO ts=2020-09-18T02:21:20.527922Z app=device-virtual source=device.go:72 msg="Added device: Random-Boolean-Device"
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4078226]
> 
> goroutine 48 [running]:
> sync.(*WaitGroup).state(...)
>         /usr/local/go/src/sync/waitgroup.go:33
> sync.(*WaitGroup).Add(0x0, 0x1)
>         /usr/local/go/src/sync/waitgroup.go:54 +0x26
> github.com/edgexfoundry/device-sdk-go/internal/autoevent.(*executor).Run(0xc00014c000, 0x0, 0x0, 0x0, 0xc000245240)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.2.4-dev.12/internal/autoevent/executor.go:41 +0x5f
> created by github.com/edgexfoundry/device-sdk-go/internal/autoevent.triggerExecutors
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.2.4-dev.12/internal/autoevent/manager.go:94 +0x27f
> 
> Process finished with exit code 2


When launching the device-virtual service for a second time:
> level=INFO ts=2020-09-18T02:22:19.460243Z app=device-virtual source=config.go:193 msg="Loaded configuration from ./res/configuration.toml"
> level=ERROR ts=2020-09-18T02:22:19.461018Z app=device-virtual source=logger.go:74 msg="logTarget cannot be blank, using stdout only"
> level=INFO ts=2020-09-18T02:22:19.461033Z app=device-virtual source=config.go:304 msg="Using local configuration from file (0 envVars overrides applied)"
> level=INFO ts=2020-09-18T02:22:19.461049Z app=device-virtual source=httpserver.go:99 msg="Web server starting (localhost:49990)"
> level=INFO ts=2020-09-18T02:22:19.461121Z app=device-virtual source=init.go:144 msg="Check Metadata service's status by ping..."
> level=INFO ts=2020-09-18T02:22:19.461189Z app=device-virtual source=init.go:144 msg="Check Data service's status by ping..."
> level=INFO ts=2020-09-18T02:22:19.46521Z app=device-virtual source=init.go:68 msg="Service clients initialize successful."
> level=INFO ts=2020-09-18T02:22:19.466672Z app=device-virtual source=service.go:153 msg="Device Service device-virtual exists"
> panic: interface conversion: interface is nil, not models.ProtocolDiscovery
> 
> goroutine 1 [running]:
> github.com/edgexfoundry/device-sdk-go/internal/container.ProtocolDiscoveryFrom(...)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.2.4-dev.12/internal/container/deviceservice.go:26
> github.com/edgexfoundry/device-sdk-go/internal/autodiscovery.BootstrapHandler(0x46fe160, 0xc0001aef40, 0xc000029300, 0xbfd12456db624cb8, 0x1c15c5, 0x49d03a0, 0x6fc23ac00, 0x3b9aca00, 0xc0001ad280, 0x1)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.2.4-dev.12/internal/autodiscovery/autodiscovery.go:26 +0xa6
> github.com/edgexfoundry/go-mod-bootstrap/bootstrap.RunAndReturnWaitGroup(0x46fe160, 0xc0001aef40, 0xc000181060, 0x4703760, 0xc000172b60, 0x4646c0b, 0xe, 0x46488c6, 0x12, 0x47013c0, ...)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/go-mod-bootstrap@v0.0.37/bootstrap/bootstrap.go:150 +0x648
> github.com/edgexfoundry/go-mod-bootstrap/bootstrap.Run(0x46fe160, 0xc0001aef40, 0xc000181060, 0x4703760, 0xc000172b60, 0x4646c0b, 0xe, 0x46488c6, 0x12, 0x47013c0, ...)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/go-mod-bootstrap@v0.0.37/bootstrap/bootstrap.go:175 +0x149
> github.com/edgexfoundry/device-sdk-go/pkg/service.Main(0x4646c0b, 0xe, 0x464d6cf, 0x1a, 0x45f9260, 0xc00007c5f0, 0x46fe160, 0xc0001aef40, 0xc000181060, 0xc0001b8540, ...)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.2.4-dev.12/pkg/service/main.go:53 +0x765
> github.com/edgexfoundry/device-sdk-go/pkg/startup.Bootstrap(0x4646c0b, 0xe, 0x464d6cf, 0x1a, 0x45f9260, 0xc00007c5f0)
>         /Users/felix/go/pkg/mod/github.com/edgexfoundry/device-sdk-go@v1.2.4-dev.12/pkg/startup/bootstrap.go:19 +0x117
> main.main()
>         /Users/felix/Git/FelixTing/device-virtual-go/cmd/main.go:21 +0x72
> 
> Process finished with exit code 2

**Comment by cloudxxx8** [comment]
@FelixTing this issue is just fixed in device-sdk-go v1.2.4-dev.12

**Comment by cloudxxx8** [comment]
there is an additional issue @hahattan is working on.  please hold the review.

**Comment by cloudxxx8** [comment]
@FelixTing The issue should be resolved.  Please help verify again.

File: go.mod (hunk 1 above)
Comment (cloudxxx8) [reply go.mod]: @lenny-intel I addressed your change request. please help review again.  thanks.

**Review by FelixTing — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
@lenny-intel could you please review this PR again?

**Review by lenny-intel — APPROVED** [review]
